### PR TITLE
Move snmp_traps config under network_devices

### PR DESCRIPTION
**Pull request by FlorianVeaux**
Currently targetting a non-main branch, waiting for this PR to get merged: https://github.com/DataDog/datadog-agent/pull/11644

### CODE REVIEW

**Review by djmitche — APPROVED** [review]
Very thorough of you to even update the tests for pkg/util/scrubber :)

**Review by maycmlee — APPROVED** [review]
👍  for docs.

File: pkg/snmp/traps/config.go [hunk 1]
```diff
@@ -46,14 +47,18 @@ type Config struct {
...
 func ReadConfig(agentHostname string) (*Config, error) {
 	var c Config
-	err := config.Datadog.UnmarshalKey("snmp_traps_config", &c)
+	err := config.Datadog.UnmarshalKey("network_devices.snmp_traps", &c)
 	if err != nil {
 		return nil, err
 	}
 
+	if !c.Enabled {
+		return nil, errors.New("traps listener is disabled")
```
Comment (AlexandreYang) [inline pkg/snmp/traps/config.go:56]: **❓ question:** ‏This is not supposed to happen right ?

File: pkg/util/scrubber/default_test.go [hunk 2]
```diff
@@ -285,53 +285,65 @@ func TestSNMPConfig(t *testing.T) {
 		`   community_string: ********`)
 	assertClean(t,
 		`
-snmp_traps_config:
-  community_strings:
-    - 'password1'
-    - 'password2'
+network_devices:
```
Comment (AlexandreYang) [inline pkg/util/scrubber/default_test.go:288]: I think we should still keep one/two tests for older `snmp_traps_config.community_strings` to make sure we maintain backward compatibility for scrubbing community string set with `snmp_traps_config.community_strings`.

The cost is low to do so and can avoid leaking community strings (e.g. users still having legacy config in newer versions of Agent).

File: pkg/snmp/traps/config.go (hunk 1 above)
Comment (FlorianVeaux) [reply pkg/snmp/traps/config.go:56]: In theory not at the moment, this function is not called if the traps listener is disabled. I'm ok to remove if you prefer

File: pkg/util/scrubber/default_test.go (hunk 2 above)
Comment (FlorianVeaux) [reply pkg/util/scrubber/default_test.go:288]: Good idea, I did not think of this 👍

**Review by AlexandreYang — APPROVED** [review]
LGTM 👍

**Pull request merged**